### PR TITLE
feat: Third Party Beneficiaries Integration

### PR DESCRIPTION
**Pull request by dilpreet2028**
Demo:

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `MifosStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: app/src/main/java/org/mifos/selfserviceapp/presenters/BeneficiaryApplicationPresenter.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                                           @ActivityContext Context context) {
+        super(context);
+        this.dataManager = dataManager;
+        subscription = new CompositeSubscription();
+    }
+
+    @Override
+    public void detachView() {
+        super.detachView();
+        subscription.unsubscribe();
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/presenters/BeneficiaryApplicationPresenter.java]: @dilpreet96 use ``subscription.clear()`` instead

File: app/src/main/java/org/mifos/selfserviceapp/presenters/BeneficiaryDetailPresenter.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    public BeneficiaryDetailPresenter(DataManager manager, @ActivityContext Context context) {
+        super(context);
+        this.manager = manager;
+        subscription = new CompositeSubscription();
+    }
+
+    @Override
+    public void detachView() {
+        super.detachView();
+        subscription.unsubscribe();
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/presenters/BeneficiaryDetailPresenter.java]: @dilpreet96 use ``subscription.clear()`` instead

File: app/src/main/java/org/mifos/selfserviceapp/presenters/BeneficiaryListPresenter.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    public BeneficiaryListPresenter(DataManager dataManager, @ActivityContext Context context) {
+        super(context);
+        this.dataManager = dataManager;
+        subscription = new CompositeSubscription();
+    }
+
+    @Override
+    public void detachView() {
+        super.detachView();
+        subscription.unsubscribe();
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/presenters/BeneficiaryListPresenter.java]: @dilpreet96 use ``subscription.clear()`` instead

File: app/src/main/java/org/mifos/selfserviceapp/ui/adapters/BeneficiaryListAdapter.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+ */
+
+public class BeneficiaryListAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
+
+    private Context context;
+
+    private List<Beneficiary> beneficiaryList;
+
+    @Inject
+    public BeneficiaryListAdapter(@ActivityContext Context context) {
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/ui/adapters/BeneficiaryListAdapter.java:33]: Good job for using ``@ActivityContext Context context``

File: app/src/main/java/org/mifos/selfserviceapp/ui/adapters/BeneficiaryListAdapter.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+    public void setBeneficiaryList(List<Beneficiary> beneficiaryList) {
+        this.beneficiaryList = beneficiaryList;
+        notifyDataSetChanged();
+    }
+
+    public class ViewHolder extends RecyclerView.ViewHolder {
+
+        @BindView(R.id.tv_beneficiary_name)
+        TextView tvName;
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/ui/adapters/BeneficiaryListAdapter.java:66]: Give one line space between this and below variable

File: app/src/main/java/org/mifos/selfserviceapp/ui/fragments/BeneficiaryApplicationFragment.java [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+
+/**
+ * Created by dilpreet on 16/6/17.
+ */
+
+public class BeneficiaryApplicationFragment extends BaseFragment implements
+        BeneficiaryApplicationView, AdapterView.OnItemSelectedListener {
+
+    @BindView(R.id.sp_account_type)
+    Spinner spAccountType;
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/ui/fragments/BeneficiaryApplicationFragment.java:45]: @dilpreet96 leave one line space between the variables, for example you can check other classes

File: app/src/main/java/org/mifos/selfserviceapp/ui/fragments/BeneficiaryApplicationFragment.java [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+
+/**
+ * Created by dilpreet on 16/6/17.
+ */
+
+public class BeneficiaryApplicationFragment extends BaseFragment implements
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/ui/fragments/BeneficiaryApplicationFragment.java:41]: @dilpreet96 what you think about class name like this ``AddBeneficiaryFragment``?

File: app/src/main/java/org/mifos/selfserviceapp/ui/fragments/BeneficiaryDetailFragment.java [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+import butterknife.ButterKnife;
+
+/**
+ * Created by dilpreet on 15/6/17.
+ */
+
+public class BeneficiaryDetailFragment extends BaseFragment implements BeneficiaryDetailView {
+
+    @BindView(R.id.tv_beneficiary_name)
+    TextView tvName;
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/ui/fragments/BeneficiaryDetailFragment.java:37]: Leave one line gap

File: app/src/main/java/org/mifos/selfserviceapp/ui/fragments/BeneficiaryListFragment.java [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+
+/**
+ * Created by dilpreet on 14/6/17.
+ */
+
+public class BeneficiaryListFragment extends BaseFragment implements RecyclerItemClickListener.
+        OnItemClickListener, SwipeRefreshLayout.OnRefreshListener, BeneficiariesView {
+
+    @BindView(R.id.rv_beneficiaries)
+    RecyclerView rvBeneficiaries;
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/ui/fragments/BeneficiaryListFragment.java:39]: Leave one line gap

File: app/src/main/java/org/mifos/selfserviceapp/ui/views/BeneficiaryApplicationView.java [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+
+/**
+ * Created by dilpreet on 16/6/17.
+ */
+
+public interface BeneficiaryApplicationView extends MVPView {
+
+    void showUserInterface();
+
+    public void showBeneficiaryTemplate(BeneficiaryTemplate beneficiaryTemplate);
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/ui/views/BeneficiaryApplicationView.java]: interface method by default are public, I don't think So you need to mention.

File: app/src/main/java/org/mifos/selfserviceapp/ui/views/BeneficiaryDetailView.java [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+
+import org.mifos.selfserviceapp.ui.views.base.MVPView;
+
+/**
+ * Created by dilpreet on 16/6/17.
+ */
+
+public interface BeneficiaryDetailView extends MVPView {
+
+    public void showUserInterface();
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/ui/views/BeneficiaryDetailView.java]: interface method by default are public, I don't think So you need to mention.

File: app/src/main/res/layout/fragment_beneficiary_application.xml [hunk 12]
```diff
@@ -0,0 +1,106 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<RelativeLayout  xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:layout_margin="@dimen/default_margin">
+    <ScrollView
```
Comment (therajanmaurya) [inline app/src/main/res/layout/fragment_beneficiary_application.xml]: @dilpreet96 please use NestedScrollView

File: app/src/main/java/org/mifos/selfserviceapp/ui/fragments/BeneficiaryApplicationFragment.java (hunk 7 above)
Comment (dilpreet2028) [reply app/src/main/java/org/mifos/selfserviceapp/ui/fragments/BeneficiaryApplicationFragment.java:41]: @therajanmaurya this fragment allows to add and update a beneficiary due to which I named it BeneficiaryApplicationFragment

Comment (therajanmaurya) [reply app/src/main/java/org/mifos/selfserviceapp/ui/fragments/BeneficiaryApplicationFragment.java:41]: ohh Sorry my bad, It's ok then😜

**Comment by dilpreet2028** [comment]
@therajanmaurya changes made

**Pull request merged**